### PR TITLE
[ML] Improve cleanup for DF Analytics HLRC tests

### LINKED ISSUE #56097: [ML] Make data frame analytics cleanup in HLRC tests more robust
The cleanup for data frame analytics jobs in the HLRC tests assumes the tests worked perfectly.  In the event that a test fails the cleanup also fails leading to many subsequent tests failing.

The offending method is https://github.com/elastic/elasticsearch/blob/5fa36dad0b8503bef3d91173bb342da555445c26/client/rest-high-level/src/test/java/org/elasticsearch/client/MlTestStateCleaner.java#L108

It needs to be made more like the cleanup methods for anomaly detection jobs and datafeeds above it in the same file that force stop/close/delete in the event of the happy path not working.

An example of a build where a cleanup failed is https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+multijob+fast+part1/5345/consoleText. You can see the initial ML test failure followed by the failure to cleanup and then many subsequent tests failing due to the left-behind data frame analytics task.

### PR DESCRIPTION
**Pull request by dimitris-athanasiou**
Adds the step of stopping all data frame analytics before
deleting them to the cleanup of the corresponding HLRC tests.

Closes #56097

### CODE REVIEW

**Review by droberts195 — APPROVED** [review]
LGTM

File: client/rest-high-level/src/test/java/org/elasticsearch/client/MlTestStateCleaner.java [hunk 1]
```diff
@@ -106,10 +107,28 @@ private void closeAllJobs() {
     }
 
     private void deleteAllDataFrameAnalytics() throws IOException {
+        stopAllDataFrameAnalytics();
```
Comment (przemekwitek) [inline client/rest-high-level/src/test/java/org/elasticsearch/client/MlTestStateCleaner.java:110]: Do we need to stop DF analytics in `MlRestTestStateCleaner` as well?

Comment (dimitris-athanasiou) [reply client/rest-high-level/src/test/java/org/elasticsearch/client/MlTestStateCleaner.java:110]: Good point. Will do.

Comment (dimitris-athanasiou) [reply client/rest-high-level/src/test/java/org/elasticsearch/client/MlTestStateCleaner.java:110]: Done in https://github.com/elastic/elasticsearch/pull/56101/commits/5fab34a19b6fb62fc5fc35c1dcfd5073d9ebe884

**Review by przemekwitek — APPROVED** [review]
LGTM

**Pull request merged**